File: python/프로그래머스/요격시스템.py
def solution(targets):
    answer = 0
    sorted_targets = sorted(targets, key=lambda x: x[1])
    

    e = 0
    for target in sorted_targets:
        if target[0] >= e:
            answer += 1
            e = target[1]
        
    return answer 

File: java/요격시스템.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        PriorityQueue<int[]> q = new PriorityQueue<>((a,b)-> a[1]-b[1]);
        for(int[] target: targets){
            q.offer(target);
        }
        
        int lastIdx = q.peek()[1];
        while(!q.isEmpty()) {
            int[] current = q.poll();
            if (q.isEmpty()) {
                answer += 1;
                break;
            }
            int[] next = q.peek();
            
            if (lastIdx > next[0]) {
                continue;
            }
            answer += 1;
            lastIdx = next[1];
        }
        return answer;
    }
}
